devtools/server/actors/replay/annotations/contentScript.js

```diff
@@ -0,0 +1,35 @@
+/* global chrome */
+
+'use strict';
+
+let window;
+
+
+function initialize(dbgWindow, RecordReplayControl) {
+  window = dbgWindow.unsafeDereference();
+
+  window.wrappedJSObject.__RECORD_REPLAY_ANNOTATION_HOOK__ =
+    (source, message) => {
+      if (!source || typeof source !== "string") {
+        window.console.error("Replay annotations source must be a string");
+        return false;
+      }
+
+      if (message && typeof message !== "object") {
+        window.console.error("Replay annotation message must be an object if set");
+        return false;
+      }
+
+      RecordReplayControl.onAnnotation(
+        "generic",
+        JSON.stringify({
+          source,
+          message
+        })
+      );
```

Comment:
Why use a "generic" kind for the annotations added here when we could just pass in the kind to the hook?  That seems to add unnecessary complexity.  If the concern is that random scripts would have access to this, I don't think that's much of a problem as if some page wants to mess up their recording with random annotations for existing kinds that's its own problem, and it's easy to see where a given annotation was added by time traveling to its bookmark.

Reply:
My instinct is always to reserve space for internal use when exposing userland methods so we don't have to worry about breaking user customizations when we want to add features later.

That said, I'm not sure this design best suits that purpose. Maybe reserving the `replay-` prefix for the kind would be sufficient.